src/pdsfile/pdscache.py

```
@@ -795,7 +795,7 @@ def set_multi(self, mydict, lifetime=None):
         # Retrieve lifetimes from cache if necessary
         if lifetime is None and nonlocal_keys:
             nonlocal_dict = self.mc.get_multi(nonlocal_keys)
-            for (key, tuple) in nonlocal_dict:
+            for (key, tuple) in nonlocal_dict.items():
```

Ruff (0.15.21)

[error] Variable `tuple` is shadowing a Python builtin

(A001)